### PR TITLE
EEP: Obsolete Execution as a Command

### PR DESCRIPTION
**Pull request by dmikurube**
This is an EEP about the plan to deprecate the `embulk` command.

Read the rendered version at:

https://github.com/embulk/embulk/blob/eep-obsolete-execution-as-a-command/docs/eeps/draft-obsolete-execution-as-a-command.md

↓ Merged

https://github.com/embulk/embulk/blob/master/docs/eeps/eep-0012.md

### CODE REVIEW

File: docs/eeps/draft-obsolete-execution-as-a-command.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Unfortunately, they differ from each other. Different versions have different command-line options obviously. Furthermore, even within the same version, different distributions may have different command-line options and behaviors.
+
+As discussed in [EEP-8: Milestones to Embulk v1.0, and versioning strategies to follow](./eep-0008.md) and [EEP-11: Moving Forward Together with Modern Java](./eep-0011.md), Embulk would target a range of Java versions, not just a single version at a time. Also, needless to say, Embulk should not limit the Java distributions it runs on.
+
+The embedded shell script (and batch file) would still need to cover those variations of Java versions and distributions, but there are too many Java variations in the world to cover, and shell scripts (and batch files) are not so powerful to cover such complexity. On the other hand, depending on another scripting runtime (such as Ruby or Python) would add another complex dependency to the Embulk ecosystem.
+
+Embulk's use-cases in the wild
+-------------------------------
+
```
Comment (hiroyuki-sato) [inline docs/eeps/draft-obsolete-execution-as-a-command.md:46]: Is this doc still a work in progress? It seems some sentences will be added in this part later.

**Comment by dmikurube** [comment]
@hiroyuki-sato Sorry for confusing! It's now ready for review.

When it is approved and merged, it's status would be `Accepted`. It would be updated to `Final` eventually when the `embulk` command is actually removed.

File: docs/eeps/draft-obsolete-execution-as-a-command.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+The `java`-based commands would also benefit users by giving them more explicit control and visibility over their Java setup. Users will have explicit control over which Java version to run Embulk on, and which Java options to set for Embulk.
+
+### Announcement
+
+Even though we don't technically need a special migration period, users would still need a clear announcement before the `embulk` command is removed, and a grace period before it is enforced.
+
+First, [a GitHub Issue #1496: "Deprecate running Embulk by a single 'embulk' command"](https://github.com/embulk/embulk/issues/1496) has been filed since February 2022, and we have received no complaints there as of September 2024.
+
+Second, since [Embulk v0.10.36](https://github.com/embulk/embulk/releases/tag/v0.10.36) as of June 2022, Embulk has started to display the following warning message when run with the `embulk` command .
```
Comment (hiroyuki-sato) [inline docs/eeps/draft-obsolete-execution-as-a-command.md:80]: `command .` -> `command.` remove space.

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍

**Comment by dmikurube** [comment]
Thanks! Merging this as EEP-12 as `Accepted` (not `Final`) as commented.

**Pull request merged**